src/gui/editorwidgets/qgsdoublespinbox.cpp

```diff
@@ -46,6 +46,19 @@ QgsDoubleSpinBox::QgsDoubleSpinBox( QWidget *parent )
   connect( this, SIGNAL( valueChanged( double ) ), this, SLOT( changed( double ) ) );
 }
 
+void QgsDoubleSpinBox::setBackgroundColor( const QColor *c )
```

Comment:
Please make this take a const reference, not a pointer, ie `void setBackgroundColor( const QColor& color = QColor() );` You can use `if ( color.isValid() )` to test whether the color has been set